opm/parser/eclipse/EclipseState/Schedule/Action/ActionAST.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+  Copyright 2018 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef ActionAST_HPP
+#define ActionAST_HPP
+
+#include <string>
+#include <vector>
+#include <memory>
+
+namespace Opm {
+
+class ActionContext;
+class ASTNode;
+
+
+
+class ActionAST{
+public:
+    ActionAST() = default;
+    explicit ActionAST(const std::vector<std::string>& tokens);
+    bool eval(const ActionContext& context, std::vector<std::string>& matching_wells) const;
+private:
+    /*
+      The use of std::shared_ptr<> here is to be able to create this class with
+      only a forward declaration of the ASTNode class. std::unique_ptr<> will
+      not cut it
```

Comment:
I don't understand this comment.  It's entirely reasonable to have a `unique_ptr` to an incomplete type.  If that unique pointer is a member of a class, then you typically need to implement a user-defined destructor for the class, but this comment is a bit too restrictive.

Reply:
OK  - I wanted to avoid creating a user defined destructor - I'll update the comment.